libenkf/include/ert/enkf/site_config.h

```diff
@@ -101,6 +103,11 @@ typedef struct site_config_struct site_config_type;
   mode_t                   site_config_get_umask( const site_config_type * site_config );
 
   site_config_type       * site_config_alloc_empty();
+  site_config_type       * site_config_alloc_default();
```

Comment:
Ideally I guess it should be possible to create a valid and usable `site_config` instance without any file-parsing; but I do not think that is yet the case? As long as a site configuration *file* is actually needed I think it is a better idea to let the `site_config_alloc`function take the configuration file as input, and remove `site_config_set_config_file( )` function.